src/main/java/com/puppycrawl/tools/checkstyle/filters/IntMatchFilterElement.java

```diff
@@ -53,11 +53,7 @@ public final int hashCode() {
 
     @Override
     public final boolean equals(Object object) {
-        if (object instanceof IntMatchFilterElement) {
-            final IntMatchFilterElement other = (IntMatchFilterElement) object;
-            return matchValue == other.matchValue;
-        }
-        return false;
+        return object instanceof IntMatchFilterElement other && matchValue == other.matchValue;
```

Comment:
`[ERROR] [checkstyle] [ERROR] C:\projects\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\filters\IntMatchFilterElement.java:56:62: Condition with && at line 56 position 61 need optimization. All method calls are advised to move to end of logic expression. [LogicConditionNeedOptimization]`

please create issue on this Check in our repo, it is false positive.
after issue is created, please suppress violation and link to issue in comment `until ....`